messages/teleporter/message_manager.go

```diff
@@ -323,3 +294,40 @@ func (m *messageManager) parseTeleporterMessage(warpMessageID ids.ID, warpPayloa
 	}
 	return teleporterMessage, nil
 }
+
+func (m *messageManager) getTeleporterMessenger(destinationClient vms.DestinationClient) (*teleportermessenger.TeleporterMessenger, error) {
+	client, ok := destinationClient.Client().(ethclient.Client)
```

Comment:
Does this casting work for the c-chain client?

Reply:
It should, since we initialize the C-Chain client using `subnet-evm`'s `ethclient`. This shouldn't be a problem since we're not using the client to fetch blocks, but we should probably rethink this.

Reply:
Yeah, for now I think `destinationClient.Client()` should return an `ethclient.Client`. At least this way when we need to add a new client type, we will know we'll need to make a change here.

Reply:
same as below